### ISSUE TITLE
Failed to find assets path for "Frameworks/App.framework/flutter_assets"

### ISSUE THREAD

**Issue by dpedrinha** [flutter/flutter#52084]
## Steps to Reproduce

I was having inconsistent results with flutter auth using passwordless email authentication on iOS. A few users were not getting redirected to the app or the app was not finishing the sign in process. Since it was inconsistent, because most users could login with no problem and I couldn't replicate the problem on my end, I decided to update everything to the latest version.
Since updating my dependencies I couldn't compile anymore, I was getting #35670 until I found versions that are now compiling but showing this error:

```bash
Xcode build done.                                           379,5s
        path: satisfied (Path is satisfied), interface: en0
Failed to find assets path for "Frameworks/App.framework/flutter_assets"
Configuring the default Firebase app...
Configured the default Firebase app __FIRAPP_DEFAULT.
[VERBOSE-2:engine.cc(127)] Engine run configuration was invalid.
[VERBOSE-2:shell.cc(437)] Could not launch engine with configuration.
        path: satisfied (Path is satisfied), interface: en0             
Syncing files to device iPhone 11 Pro...                                
    dismiss = 1;                                                
    "is_weak_match" = 1;                                                                                           
}                                                                                                                  
Syncing files to device iPhone 11 Pro...                        17.068ms (!)
```
The app runs but there's only a white screen.

Here is my pubspec.yaml (minus the private info):

```yaml
version: 6.2.5+220

environment:
  #sdk: ">=2.1.0 <3.0.0"
  sdk: ">=2.2.2 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.2

  shared_preferences: ^0.5.6+2
  provider: ^4.0.4
  #provider: ^3.2.0 #versão mais antiga
  #countdown_flutter: ^0.1.1
  numberpicker: ^1.1.0
  flutter_local_notifications: ^1.2.0+4
  notification_permissions: ^0.4.4
  configurable_expansion_tile: ^1.0.0
#  firebase_core: any
#  firebase_auth: any
#  firebase_dynamic_links: any 
#  cloud_firestore: any
#  firebase_analytics: any
  firebase_core: 0.4.4
  firebase_auth: 0.15.0 #0.15.5 0.15.0
  firebase_dynamic_links: ^0.5.0+9 
  cloud_firestore: 0.13.4
  firebase_analytics: ^5.0.11
  path_provider: ^1.6.4
  xml: ^3.5.0
  photo_view: ^0.9.2
  google_sign_in: ^4.1.1
  googleapis_auth: ^0.2.11+1
  googleapis: ^0.54.0
  package_info: ^0.4.0+14
#  giffy_dialog: ^1.3.0
#  http: ^0.12.0+2
  flutter_localizations:
    sdk: flutter
  #flutter_cupertino_localizations: ^1.0.1
  devicelocale: ^0.2.1
  flutter_email_sender: ^2.2.2
  #duration: ^2.0.8

dev_dependencies:
  flutter_test:
    sdk: flutter


# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  assets:
    - images/
    - images/point_images/backimg/
    - images/point_images/midimg/
    - images/point_images/frontimg/
    - jsonfiles/

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  fonts:
    - family:  AAIcons
      fonts:
        - asset: fonts/AAIcons.ttf
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

```

I tried stable, master and beta channels with the same result.
I also already tried all three channels with both Build Settings, Legacy and New. Same result.

On every attempt I try the following sequence:

```console
flutter clean
cd ios
pod deintegrate
pod install
cd ..
flutter build ios
flutter run
```
I'm pretty sure something got messed up on one of the versions of flutterfire plugins and flutter clean and pod deintegrate is not working.
This is a production app, the only things I changed were the plugins versions and flutter channels when this happened. I tried rolling back pubspec.yaml versions from the last working version from my git repo and this new problem persists.
Any help is appreciated.

**Expected results:** 
App working
**Actual results:** 
White screen on app
<details>
  <summary>logs</summary>

```bash
Xcode build done.                                           379,5s
        path: satisfied (Path is satisfied), interface: en0
Failed to find assets path for "Frameworks/App.framework/flutter_assets"
Configuring the default Firebase app...
Configured the default Firebase app __FIRAPP_DEFAULT.
[VERBOSE-2:engine.cc(127)] Engine run configuration was invalid.
[VERBOSE-2:shell.cc(437)] Could not launch engine with configuration.
        path: satisfied (Path is satisfied), interface: en0             
Syncing files to device iPhone 11 Pro...                                
    dismiss = 1;                                                
    "is_weak_match" = 1;                                                                                           
}                                                                                                                  
Syncing files to device iPhone 11 Pro...                        17.068ms (!) 
```

</summary>

```console
No issues found! (ran in 6.6s)
```

```console
Unable to find any JVMs matching version "(null)".
No Java runtime present, try --request to install.
Unable to find any JVMs matching version "(null)".
No Java runtime present, try --request to install.
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale pt-BR)
    • Flutter version 1.12.13+hotfix.8 at /Users/myuser/dev/flutter
    • Framework revision 0b8abb4724 (3 weeks ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/myuser/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /usr/bin/java
    ✗ Could not determine java version

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).

[✓] VS Code (version 1.42.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (1 available)
    • iPhone 11 Pro • 35EAE23A-04E2-4D0C-8176-02B7D2AB5354 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

! Doctor found issues in 2 categories.
```

**Comment by iapicca** [comment]
Hi @dpedrinha 

if you believe the issue is related to a FlutterFire plugin,
please open the issue in the dedicated [github](https://github.com/FirebaseExtended/flutterfire/issues).

if you think instead that it's caused by flutter itself
can you please provide your `flutter run --verbose` 
and if possible a reproducible minimal code sample.
Thank you

**Comment by dpedrinha** [comment]
Hello @iapicca 
I'm having multiple problems. The problem with flutterfire version is already issued, but I believe this one is on flutter.
Here's a gist with the flutter run --verbose
[https://gist.github.com/dpedrinha/ea91feabc8467d88c739b232bdbbc721](https://gist.github.com/dpedrinha/ea91feabc8467d88c739b232bdbbc721)
Thanks for looking into it.

**Comment by dpedrinha** [comment]
Found this on the --verbose. Maybe it helps...

```
fatal error: file '/Users/myuser/dev/apps/myappname/ios/Flutter/Flutter.framework/Headers/FlutterTexture.h' has been modified since the
                     precompiled header
```

**Comment by dgurudot** [comment]
Having same issue which is tracked here. No workaround or resolution so far

https://github.com/flutter/flutter/issues/33121

**Comment by dpedrinha** [comment]
Master channel is working after #51453 
This should go to stable channel ASAP because it's definitely not stable.

**Comment by jmagman** [comment]
You created this issue (tested on the master channel) 2 days after https://github.com/flutter/flutter/pull/51453 was merged.

**Comment by jmagman** [comment]
Come to think of it, based on the timing, it's actually possible this was _caused_ by https://github.com/flutter/flutter/pull/51453 because of that migration.  Before you saw this error, were you on master at some point, then switched back to beta or stable?

The project needed to be adjusted to handle the way Xcode 11.4 links and embeds frameworks.  If you are on beta or stable and see this, does following the instructions in https://flutter.dev/docs/development/ios-project-migration fix it?

**Comment by dgurudot** [comment]
Switching back to stable and I am trying migration steps given above.

**Comment by dgurudot** [comment]
https://flutter.dev/docs/development/ios-project-migration

These migration steps helped to solve the issue on stable branch. Thx so much for quick response.

**Comment by dpedrinha** [comment]
> Come to think of it, based on the timing, it's actually possible this was _caused_ by #51453 because of that migration. Before you saw this error, were you on master at some point, then switched back to beta or stable?
> 
> @dgurudot @dpedrinha The project needed to be adjusted to handle the way Xcode 11.4 links and embeds frameworks. If you are on beta or stable and see this, does following the instructions in https://flutter.dev/docs/development/ios-project-migration fix it?

@jmagman yes I was on stable because Master was having a problem not showing bold characters on iOS. And yes, I did test master and beta before opening the issue. But it is possible that it took me a day or two to do so since first I was looking for an open issue or SO questions regarding the problem. Usually I try hard and look for days before opening an issue here. As I described, I tried multiple `flutter-fire` versions in combination with `flutter channels` and Xcode `build settings` and all that took me days of `flutter clean`, `flutter upgrade` and `flutter run` lol.

And for some reason it was only after @dgurudot replied here that I found his issue. 

I'll try the fix later. Thanks for all the help.

**Comment by dpedrinha** [comment]
Sorry guys but I'm not having any free time to test it.

`Flutter clean, upgrade and run` take about 20 minutes on my Macbook Air and I need to get back on track since my app is in production and I was stalled for about two weeks because of this problem.

But I'm sure we can take @dgurudot word for it.

**Comment by balajiks-dev** [comment]
Still not solved. Kindly resolve this issue.

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/51453 will be in the beta channel any second.  If you switch from beta, dev, or master back to stable, you may hit this issue.  Please try the migration steps at https://flutter.dev/docs/development/ios-project-migration.

@balajiks-dev What's the output of `flutter doctor -v`?  You tried the migration steps and are still hitting it?

**Comment by PROGrand** [comment]
I switched from master to stable and meet this issue. Reinstalling of flutter does not help. But how I fixed it:
1) remove ios/Flutter
2) recreated projects using something like this: "flutter create -i objc -a java --project-name MyProjectName ."

**Comment by jmagman** [comment]
If you are seeing this after switching from master, dev, or beta channels back to stable, it is likely because Flutter tool has migrated your project to [fix the way Xcode 11.4 now links and embeds frameworks](https://github.com/flutter/flutter/issues/50568) in v1.15.4 and later.  When you switch back to stable, the tool hasn't been updated to handle this.  Please follow the instructions in https://flutter.dev/docs/development/ios-project-migration to get it working again.  

I'm going to keep this issue open for visibility, and lock it for now to avoid "me too!" comments and misleading suggestions.  Please file a new issue and completely fill out the template if you followed the migration instructions and are still seeing similar error messages.

**Comment by jmagman** [comment]
Flutter 1.17.0 is now in stable, so this should be fixed on all channels.  If you are seeing this issue, please run `flutter upgrade`.
